plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -527,7 +537,29 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
 
         TupleDomain<ColumnHandle> oldDomain = handle.getConstraint();
         TupleDomain<ColumnHandle> newDomain = oldDomain.intersect(TupleDomain.withColumnDomains(supported));
-        if (oldDomain.equals(newDomain)) {
+
+        ConnectorExpression oldExpression = constraint.getExpression();
+        ConnectorExpression newExpression = constraint.getExpression();
+        Map<String, String> newRegexes = new HashMap<>(handle.getRegexes());
+        if (constraint.getExpression() instanceof Call) {
```

Comment:
`constraint.getExpression()` should be decomposed into conjuncts we should handle any conjunct that is translateable (currently: is a LIKE pattern CALL that we can translate)

example where it matters

- col LIKE 'pattern' AND col LIKE 'another_pattern'
- col LIKE 'pattern' AND col2 LIKE 'another_pattern'
- col LIKE 'pattern' AND some-other-expression

Reply:
But we don't translate `LogicalExpression` yet (see https://github.com/trinodb/trino/issues/9506)

Reply:
You're right. I missed the fact we didn't add LogicalExpression to ConnectorExpression just yet.
Still connector implementation should be future-proof. More precisely, it must be possible to write future-proof connector implementation, so we need to introduce proper abstractions. "decomposed into conjuncts" seems like the right thing to have, in plugin-toolkit. See `io.trino.sql.ExpressionUtils#extractConjuncts` for ref.

Reply:
Done